cypress/dashboards/example.json

```diff
@@ -0,0 +1,210 @@
+{
+  "annotations": {
+    "list": [
+      {
+        "builtIn": 1,
+        "datasource": "-- Grafana --",
+        "enable": true,
+        "hide": true,
+        "iconColor": "rgba(0, 211, 255, 1)",
+        "name": "Annotations & Alerts",
+        "target": {
+          "limit": 100,
+          "matchAny": false,
+          "tags": [],
+          "type": "dashboard"
+        },
+        "type": "dashboard"
+      }
+    ]
+  },
+  "editable": true,
+  "fiscalYearStartMonth": 0,
+  "gnetId": null,
+  "graphTooltip": 0,
+  "id": 2,
+  "links": [],
+  "liveNow": false,
+  "panels": [
+    {
+      "datasource": null,
```

Comment:
here is probably where you need set the datasource info. Something like:

```
"datasource": "e2e-azure-data-explorer-datasource"
```

The problem is that in newer versions, it should be something like:
```
          "datasource": {
            "type": "grafana-azure-monitor-datasource",
            "uid": "Cq2BrVeGz"
          },
```
(and I have no idea how to obtain that `uid`)

But as you discovered, the import does not fail if the datasource doesn't exist. So this test is not really doing anything. I remember that @sarahzinger mentioned that we could extend this test to do something else. You can check with her.

Reply:
Hmm It sounds like a test adding a dashboard from UI would be more useful for the moment

Reply:
Oh also I could also add the datasource at the beginning of this scenario.

Reply:
Yeah since 8.3.0, the datasource name is replaced by a data source ref (with type and uid). See https://github.com/grafana/grafana/pull/33817. I believe this change is forward compatible but not backwards compatible. So if we want to e2e test grafana < 8.3, we should use the data source name as Andrés suggests. Sorry if I'm missing context here, but do you know the name of the data source while importing the dashboard? If so, it should be safe to use it.

Reply:
I think if the import doesn't fail if the datasource doesn't exist, maybe that's a bug with importDashboards that we could fix, let me know if you'd like to pair on it!

Reply:
Thanks @sarahzinger for pairing with me.

Now that I correctly reference the datasource, it fails as we expect on panel validation if that datasource was not added before.